core/src/test/scala/unit/kafka/admin/ReassignPartitionsClusterTest.scala

```diff
@@ -1103,6 +1101,78 @@ class ReassignPartitionsClusterTest extends ZooKeeperTestHarness with Logging {
     assertEquals(Seq(101), zkClient.getReplicasForPartition(tpB0))
   }
 
+  /**
+   * Validates that partitions can be created for topics not in reassignment and for the topics that are in reassignment
+   * an ReassignmentInProgressException should be thrown.cThe test creates two topics `topicName` and `otherTopicName`,
+   * the `topicName` topic undergoes partition reassignment and the test validates that during reassignment createPartitions
+   * call throws ReassignmentInProgressException `topicName` topic and for topic `otherTopicName` which is not being reassigned
+   * successfully creates partitions. Further validates that after the reassignment is complete for topic `topicName`
+   * createPartition is successful for that topic.
+   */
+  @Test
+  def shouldCreatePartitionsForTopicNotInReassignment(): Unit = {
+    startBrokers(Seq(100, 101))
+    val otherTopicName = "anyTopic"
+    val otp0 = new TopicPartition(otherTopicName, 0)
+    val otp1 = new TopicPartition(otherTopicName, 1)
+    adminClient = createAdminClient(servers)
+    createTopic(zkClient, topicName,
+      Map(otp0.partition() -> Seq(100),
+          otp1.partition() -> Seq(100)),
+      servers = servers)
+    createTopic(zkClient, otherTopicName,
+      Map(tp0.partition() -> Seq(100),
+          tp1.partition() -> Seq(100)),
+      servers = servers)
+
+    // Throttle to avoid race conditions
+    throttle(Seq(topicName), throttleSettingForSeconds(10), Map(
+      tp0 -> Seq(100, 101),
+      tp1 -> Seq(100, 101)
+    ))
+    // setup reassignment for topicName partitions
+    val move = Map(
+      tp0 -> Seq(101),
+      tp1 -> Seq(101)
+    )
+    zkClient.setOrCreatePartitionReassignment(move, ZkVersion.MatchAnyVersion)
```

Comment:
Could we use the AdminClient instead? We want to deprecate the zk API. There are a few helpers in this class to make this easier (e.g. `waitForAllReassignmentsToComplete` instead of `waitForZkReassignmentToComplete`).

Reply:
Thanks, updated the PR